### DISCUSSION TITLE
How can I use docker to develop with 0.4.0? (i.e., how can I configure where /_build/ is exposed?)

### DISCUSSION THREAD

**Discussion by knpwrs** [solidjs/solid-start#1272] [Q&A]
I use docker for my local development, and I have the primary application port exposed on my machine. Since upgrading to 0.4.0, however, the network tab shows repeated attempts to `/_build/` on a _different port_ than what the rest of the app is served from:

<img width="377" alt="image" src="https://github.com/solidjs/solid-start/assets/174864/9b7b2930-a5ab-417c-bee8-f21da23552f0">

How can I configure the port where `/_build/` is served from so I can reliably expose it from Docker and the connection can succeed? I've tried tracing the code through to vinxi but wasn't able to determine the correct parameters for my configuration.

**Comment by mehdinajafi** [comment]
Try this

In your `app.config.ts`:

``` typescript
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  vite({ router }) {
    if (router === "client") {
      return {
        server: {
          hmr: {
            port: 3001,
          },
        },
      };
    }
    return {};
  },
});

```

And in your `docker-compose.yml`:
```yaml
ports:
      - 3001:3001
```
